pydantic/main.py

```diff
@@ -56,6 +58,42 @@
 
 _object_setattr = _model_construction.object_setattr
 
+_K = TypeVar('_K')
+_V = TypeVar('_V')
+
+
+# We need a sentinel value for missing fields when comparing models
+# Models are equals if-and-only-if they miss the same fields, and since None is a legitimate value
+# we can't default to None
+# We use the single-value enum trick to allow correct typing when using a sentinel
+class _SentinelType(enum.Enum):
+    SENTINEL = enum.auto()
+
+
+_SENTINEL = _SentinelType.SENTINEL
```

Comment:
```suggestion
_SENTINEL = object()
```

I think it's best to keep it simple.